2_process/src/munge_inst_timeseries.R

```diff
@@ -55,3 +55,26 @@ aggregate_data_to_hourly <- function(inst_data,output_tz){
 
 }
 
+aggregate_data_to_daily <- function(inst_data, daily_data){
+  #' 
+  #' @description Function to aggregate instantaneous NWIS data collected at sub-hourly (e.g. 15/5/30 min) intervals to hourly min/mean/maxs
+  #'
+  #' @param inst_data a data frame containing the downloaded time series for NWIS instantaneous site. 
+  #' inst_data must include the following columns: c("Value_Inst","dateTime","site_no")
+  #' @param daily_data a data frame the downloaded daily time series of DO data. This is used so that if a site is already in 
+  #'  the daily sites, we won't do the aggregating here
+  #'
+  #' @value A data frame containing daily min, mean, and max values the parameter of interest
+  
+  
+  only_inst_data = setdiff(inst_data$site_no, daily_data$site_no)
+  
+  daily_values <- inst_data %>%
+    filter(site_no %in% only_inst_data) %>%
+    mutate(Date = as.Date(dateTime, format="%Y-%m-%d")) %>%
```

Comment:
`as.Date` will return the date associated with the time zone of `dateTime`, which is currently "UTC". For daily summaries I think we should consider using local time to aggregate the data to daily time steps (some discussion of time zones from the drb-do project is [here](https://github.com/USGS-R/drb-do-ml/issues/22). I would guess the local WSC's also report the daily data using dates based on local time.

If you agree with using local time, one way to implement that is to use helper functions in the {lubridate} package:
<pre>
dateTime_local = lubridate::with_tz(dateTime,tzone="America/New_York")
Date = lubridate::date(dateTime_local)
</pre>

Reply:
Yes, good call on aggregating in local time. I think that's the way to catch daylight savings and leap year quirks that happen in some datasets (looks like your DO data had a site like that)

Reply:
I also compared daily summaries based on UTC vs. local eastern time for a few sites that report instantaneous data as well as the daily summaries. The local time estimates were a better match, so I think it's safe to go with that approach.